daemon/cmd/root.go

```diff
@@ -49,18 +51,36 @@ var (
 	agentHive *hive.Hive
 )
 
+type DaemonCellConfig struct {
+	SkipDaemon bool
+}
+
+func (DaemonCellConfig) CellFlags(flags *pflag.FlagSet) {
+	flags.Bool("skip-daemon", false, "Skip running of the daemon, only start normal cells")
+	flags.MarkHidden("skip-daemon")
+}
+
 func init() {
 	cobra.OnInitialize(option.InitConfig(RootCmd, "cilium-agent", "cilium", Vp))
 	setupSleepBeforeFatal()
 	registerBootstrapMetrics()
 	initializeFlags()
 
+	flags := pflag.NewFlagSet("hive", pflag.ContinueOnError)
```

Comment:
This all is a bit confusing. there's `initializeFlags()` which takes no arguments, then there's the `flags` var, then there's `RootCmd.Flags()`? And commands 

Is there any way to make this simpler? I would expect that creating the Hive would make the flags Just Work.

Reply:
The reason for this is that the root, "dot-graph" and "objects" commands in cilium-agent want all the same flags, but then we have "cmdref" which shouldn't. But now that you pointed this out I can just made `cmdref` a hidden command and then we don't have `Documentation/cmdref/cilium-agenc_cmdref.md` and it doesn't matter what flags are there (unless we start adding mandatory flags at some point etc.).

Reply:
I get it now. I realized as I was walking for coffee that a Cell is different from a subcommand. Nevertheless, any way you can make flags simpler would be good; I've personally had poor experiences with other "do-it-all" cmd frameworks making it unergonomic to set flags the way you want.